### DISCUSSION TITLE
cpp/memory/smart_pointers

### DISCUSSION THREAD

**Discussion by giscus** [imarvinle/CSGuide#76] [Announcements]
# cpp/memory/smart_pointers

编程指北: 计算机系统学习指南, 操作系统, 计算机网络, C++, Java, 算法, 数据结构

https://csguide.cn/cpp/memory/smart_pointers.html

**Comment by fangtaoysu** [comment]
野指针：未初始化或已经被释放的指针被称为野指针。
准确说只有未初始化的叫野指针，已经被释放的叫悬空指针

**Comment by czydyy** [comment]
nt main() {
    std::shared_ptr<int> ptr1(new int(10)); {
    std::shared_ptr<int> ptr2 = ptr1; // 通过拷贝构造函数创建一个新的shared_ptr，此时引用计数为2
    std::cout << *ptr1 << " " << *ptr2 << std::endl; // 输出10 10 }
    // ptr2超出作用域时，所指向的内存不会被释放，因为此时ptr1仍然持有对该内存的引用
    return 0;
} 
shared_ptr的示例，少一个{}
